test/integration/features/proxy.feature

```diff
@@ -0,0 +1,81 @@
+@proxy @minishift-only
+Feature: Minishift can run behind proxy
+  As a user I can use Minishift behind a proxy.
+  INFO: Code behind this feature will start a proxy server in background
+        on port 8181 and will try to get IP of host machine automatically
+        to set MINISHIFT_HTTP_PROXY for this feature only.
+        However one can override port number or set IP manually if needed.
+        To use custom port number run this test with environmental variable
+        INTEGRATION_PROXY_CUSTOM_PORT.       
+        To use custom IP address please set INTEGRATION_PROXY_CUSTOM_IP
+        environmental variable.
+
+  Scenario: Start behind the proxy
+   Given user starts proxy server and sets MINISHIFT_HTTP_PROXY variable
+    When executing "minishift start" succeeds
+    Then proxy log should contain "Accepting CONNECT to registry-1.docker.io:443"
+     And proxy log should contain "Accepting CONNECT to auth.docker.io:443"
+
+  Scenario: User can login to the server
+   Given Minishift has state "Running"
+    When executing "oc login --username=developer --password=developer" succeeds
+    Then stdout should contain
+     """
+     Login successful
+     """
+
+  Scenario: User can create new namespace ruby for application ruby-ex
+   Given Minishift has state "Running"
+    When executing "oc new-project ruby" succeeds
+    Then stdout should contain
+     """
+     Now using project "ruby"
+     """
+
```

Comment:
What I am missing is a another simple check. When proxies are specified we actually also set the environment variables HTTP(S)_PROXY an NO_PROXY in the VM environment. So I believe that something like:

    $ minishift ssh -- echo $HTTP_PROXY

should return the configured proxy.